src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImplTest.java

```diff
@@ -354,11 +358,13 @@ public void advanceBlockchain_with_brothers_ok() throws HSMClientException, Json
         when(jsonRpcClientMock.send(buildExpectedRequest("blockchainState", VERSION_THREE)))
             .thenReturn(buildResponse(false));
 
-        BlockHeader blockHeader = mock(BlockHeader.class);
-        when(blockHeader.getFullEncoded()).thenReturn(new byte[]{});
+        List<Block> blocks = Arrays.asList(
+            TestUtils.mockBlock(1, TestUtils.createHash(1)),
+            TestUtils.mockBlock(2, TestUtils.createHash(2)),
+            TestUtils.mockBlock(3, TestUtils.createHash(3)));
```

Comment:
This is confusing to me, `mockBlock` is not adding any brother, just passing an empty list, but `advanceBlockchain` method somehow is sending a list of three brothers(same size of headers). So, this means the payload includes `parsedBlockHeaders` as brothers of themself, right?

Reply:
The Scope For This Test Is In Another Ticket, However, I Have Updated To Use `mockBlockWithBrothers`

Reply:
Fantastic.